src/iproute2_sysrepo.c

```diff
@@ -81,10 +82,15 @@ struct rtnl_handle rth = { .fd = -1 };
 sr_session_ctx_t *sr_session;
 static sr_conn_ctx_t *sr_connection;
 static sr_subscription_ctx_t *sr_sub_ctx;
-static char *iproute2_ip_modules[] = { "iproute2-ip-link", "iproute2-ip-nexthop",
-                                       "iproute2-ip-netns", NULL }; // null terminator
-
-extern char json_buffer[1024 * 1024];
+/**
+ * Struct store yang modules names, operational data path and callabck functions.
+ */
+struct yang_module {
+    const char *module; // Stores Module name
+    const char *oper_pull_path; // Stores operational pull subscription path
+} ipr2_ip_modules[] = { { "iproute2-ip-link", "/iproute2-ip-link:links/link" },
```

Comment:
why do we need `oper_pull_path` ? why not just subscribe to module `iproute2-ip-link` using the same subscription function, `sr_subscribe_config()` but rename it to `sr_subscribe()`, and this function will subscribe to running and oper_data, and assignee corresponding `callback`

Reply:
what if i want to `/iproute2-ip-link:links` or if i want to pull everything in the module?